src/basic/special.h

```diff
@@ -112,8 +112,9 @@
 #define SPECIAL_MACHINE_SLICE "machine.slice"
 #define SPECIAL_ROOT_SLICE "-.slice"
 
-/* The scope unit systemd itself lives in. */
+/* The scope units systemd itself lives in. */
 #define SPECIAL_INIT_SCOPE "init.scope"
+#define SPECIAL_INIT_WORKERS_SCOPE "init-workers.scope"
```

Comment:
hmm, instead of making this a .scope unit, maybe move pid1 into `init.scope/main/` and the workers in `init.scope/workers/`?

that way the no-processes in inner cgroups rule isn't broken either.

Reply:
I thought about that, but I am afraid that it would break things doing tracking of pid1's resources state (of which I have a few), as it changes the existing layout

Reply:
i'd be more than happy to break code that assumes that path cannot change.

Reply:
but that would mean you can't use systemctl status or show to get information about only pid1 or only the workers, because they are no longer separate scopes, no? I do need to be able to do that

Reply:
the thing is that we might acquire more and more of these things sooner or later, it doesn't scale if we put these as siblings of init.scope.

Seriously, let's do this cleanly. If not even us buy into the cgroup nesting story properly, who will?

Reply:
I do not mind how the cgroups are named, or how they are arranged positionally. I am fine with siblings at the top level, siblings under an intermediate node, or completely separate and disjoint trees. I also do not mind how they are called - it would probably be better to keep the same name for backward compat, but in the end if it is really important for you I am fine with also renaming things, and I will deal with the fallout from it.

However, I really need to be able to sensibly manage resource allocations. I have added this long explanation to the commit message accordingly:

In order to avoid paying the 50% slowdown cost, in the common case there
must always be processes ready to pick up work, so the pool cannot be
empty for long. This implies that the pool is long lived (even though
individual workers come and go), and its resources need to be sensibly
manageable. It is quite normal when there is a control plane/workers
split to be able to manage them separately, as the control plane and
the workers naturally respond to different resource allocation needs.
For this reason, the workers are spawned into a separate cgroup,
init-workers.scope, rather than init.scope. This allows setting different
CPU/Memory/IO levels and limits. For example, it would be very unwise to
set MemoryMax= on init.scope, given that means control plane processes
might get killed and cause havoc - while PID1 is immune, its short-lived
helper processes (cleanup, dissect, mount namespace tunnels, etc) would
not. Setting MemoryMax= on the workers on the other hand is a good       
safety measure. But it goes beyond that - for my use case, real world
data and metrics from my fleet tell me that PID1, after some months
of uptime, sits between 35M and 45M memory usage, and I have a limit of
48M. Workers take ~2M each, so with 5 workers by default I am already 
over my available resource pool, and if they were part of the same
cgroup it would be perpetually under pressure with processes being
killed.. So, I need to set the following resource management limits:

init.scope: MemoryHigh=48M MemorySwapMax=100M StartupMemoryHigh=infinity
init-workers.scope: MemoryHigh=20M MemorySwapMax=0 StartupMemoryHigh=100M MemoryMax=50M StartupMemoryMax=250M

This means PID1 doesn't normally get throttled and put under pressure,            
unless things go really badly and then there's some zswap to smoothen
things out, and at boot/shutdown when usage is highest there are no
limits at all, so that there are no slowdowns. And on the other hand,
the workers pool has lower overall settings and no swap, as normally
it is not intended to be that much variable, apart again from boot
and shutdown when there are 5 times more workers being spawned, and
with high ceiling safeguards in case things to really wrong so that
I don't DOS my own system. In my case memory is the most critical
resource, for others similar evaluations might apply for CPU.

These limits need to be configurable in a sensible way, as they are
routinely managed by the orchestrator system depending on what happens
on a given node, hence why a scope works best so that 'systemctl' can
be used to query current resource usage or change the limits, without
having to give permissions to manually change things under
/sys/fs/cgroup/, which would be especially bad from the security point
of view as there is a single SELinux label for the whole thing, so if 
somebody has permission to touch memory limits they can also do anything
else they want to the cgroup tree, including killing processes or moving
them around, which is definitely not nice.

Reply:
I have removed this commit entirely from this PR, so that we can get the largest change merged and done, with no new cgroup at all. I will open a follow-up that introduces the workers pool.